cluster/src/main/java/org/apache/iotdb/cluster/query/shownow/ClusterShowNowQueryExecutor.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.iotdb.db.query.dataset.ShowNowResult;
+import org.apache.iotdb.db.query.executor.ShowNowQueryExecutor;
+import org.apache.iotdb.tsfile.read.query.dataset.QueryDataSet;
+
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+
+import static org.apache.iotdb.cluster.query.ClusterPlanExecutor.*;
+
+public class ClusterShowNowQueryExecutor extends ShowNowQueryExecutor {
+  private static final Logger logger = LoggerFactory.getLogger(ClusterShowNowQueryExecutor.class);
+  private MetaGroupMember metaGroupMember;
+
+  private static ExecutorService showNowQueryPool =
+      Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+
+  public ClusterShowNowQueryExecutor(ShowNowPlan showNowPlan, MetaGroupMember metaGroupMember) {
+    super();
+    this.metaGroupMember = metaGroupMember;
+  }
+
+  @Override
+  public QueryDataSet execute(QueryContext context, ShowNowPlan showNowPlan)
+      throws MetadataException {
+    ConcurrentSkipListSet<ShowNowResult> resultSet = new ConcurrentSkipListSet<>();
+
+    ExecutorService pool =
+        new ThreadPoolExecutor(
+            THREAD_POOL_SIZE, THREAD_POOL_SIZE, 0, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+    List<Node> nodeList = metaGroupMember.getPartitionTable().getAllNodes();
+    List<Future<Void>> futureList = new ArrayList<>();
+    for (Node node : nodeList) {
+      futureList.add(
+          pool.submit(
+              () -> {
+                try {
+                  showNow(node, showNowPlan, resultSet, context);
+                } catch (Exception e) {
+                  logger.error("Cannot get show now result of {} from {}", showNowPlan, node);
+                }
+                return null;
+              }));
+    }
+
+    waitForThreadPool(futureList, pool, "showNow()");
+    List<ShowNowResult> showNowResults = applyShowNow(resultSet);
+    return new ShowNowDataSet(showNowResults, context, showNowPlan);
+  }
+
+  private void showNow(
+      Node node, ShowNowPlan showNowPlan, Set<ShowNowResult> resultSet, QueryContext context) {
+    if (node.equals(metaGroupMember.getThisNode())) {
+      showLocalNow(node, showNowPlan, resultSet, context);
+    } else {
+      showRemoteNow(node, showNowPlan, resultSet);
+    }
+  }
+
+  private void showLocalNow(
+      Node node, ShowNowPlan plan, Set<ShowNowResult> resultSet, QueryContext context) {
+    try {
+      List<ShowNowResult> localResult = null;
+      try {
+        localResult = new ShowNowUtils().getShowNowResults();
+      } catch (Exception e) {
+        e.printStackTrace();
+        logger.error("ClusterShowNowQueryExecutor showLocalNow");
```

Comment:
delete the `e.printStackTrace();` and log the exception in the log. 
The same as others.